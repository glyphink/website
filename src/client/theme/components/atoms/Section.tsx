/**
 * @file Components - Section
 * @module website/theme/components/atoms/Section
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type ForwardRefExoticComponent,
  type JSX
} from 'react'
import type SectionProps from './Section.props.mts'

/**
 * A standalone section.
 *
 * @see https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/section
 * @see {@linkcode ForwardRefExoticComponent}
 * @see {@linkcode SectionProps}
 *
 * @category
 *  atoms
 *
 * @const {ForwardRefExoticComponent<SectionProps>} Section
 */
const Section: ForwardRefExoticComponent<SectionProps> = forwardRef<
  HTMLElement,
  SectionProps
>(
  /**
   * @param {Props<SectionProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<section>` element
   */
  (props: Props<SectionProps>, ref: ForwardedRef<HTMLElement>): JSX.Element => {
    return <section {...props} ref={ref} />
  }
)

Section.displayName = 'Section'

export default Section
