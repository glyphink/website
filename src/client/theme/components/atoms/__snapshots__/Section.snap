// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/Section > should render <section> 1`] = `<section />`;
